src/MongoDB.Driver.Core/Core/Authentication/SaslAuthenticator.cs

```diff
@@ -136,8 +138,8 @@ public BsonDocument CustomizeInitialIsMasterCommand(BsonDocument isMasterCommand
             {
                 case "SCRAM-SHA-1":
```

Comment:
It seems odd that this class would contain code that applies only to one of the subclasses.  Could it be refactored so that this code lives in ScramShaAuthenticator instead?

Reply:
Hm that's an interesting question!
Prior to you asking this, `ISaslAuthenticator.CustomizeInitialIsMasterCommand` needed to call `ISaslAuthenticator.CreateStartCommand` if and only if the mechanism is SCRAM-SHA-1 or SCRAM-SHA-256.

I've included a refactor in my last set of commits so we can encapsulate the logic a bit better without exposing a new public API (leveraging C#7.2's [`private protected` modifiers](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-7.2/private-protected)).